Week-3/3-1.js

```diff
@@ -0,0 +1,22 @@
+function memoize(fn) {
+    const cache = new Map();
+    return function(...args) {
+        let key = args.join('+');
```

Comment:
This good, but some edge cases are missing.

For example:

function somefun1() {
    return 1;
}
function somefun2() {
    return 2;
}
//Create a method called memoize such that:
const memoize1 = memoize(somefun2); //then calling...
const memoize2 = memoize(somefun1); //then calling...
console.log(memoize1()); **//returns 2, which is wrong**

Build a habit of considering the edge cases while we write a programming, which is must during DSA course.